pkg/buildah/client.go

```diff
@@ -0,0 +1,138 @@
+package buildah
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/ocibuilder/ocibuilder/pkg/apis/ocibuilder/v1alpha1"
+	"github.com/ocibuilder/ocibuilder/pkg/command"
+	"github.com/sirupsen/logrus"
+)
+
+type Client struct {
+	Logger *logrus.Logger
+}
+
+func (cli Client) ImageBuild(options v1alpha1.OCIBuildOptions) (v1alpha1.OCIBuildResponse, error) {
+
+	buildFlags := []command.Flag{
+		{Name: "f", Value: options.Dockerfile, Short: true, OmitEmpty: true},
+		{Name: "storage-driver", Value: options.StorageDriver, Short: false, OmitEmpty: true},
+		{Name: "t", Value: options.Tags[0], Short: true, OmitEmpty: true},
+	}
+
+	cmd := command.Builder("buildah").Command("bud").Flags(buildFlags...).Args(options.ContextPath).Build()
+	cli.Logger.WithField("cmd", cmd).Debugln("executing build with command")
+
+	stdout, stderr, err := execute(&cmd)
+	if err != nil {
+		cli.Logger.WithError(err).Errorln("error building image...")
```

Comment:
Wrap the error error.Wrap() instead of logging the error all the time

Reply:
One note on this - error.Wrap() works but the format of the new error is hard to ready and looks like is just concatenated. I'd prefer to continue logging but only when useful fields or additional information is being added to the logs

What are your thoughts?